packages/manager/apps/pci-private-network/src/components/global-regions/GlobalRegionsDatagrid.tsx

```diff
@@ -0,0 +1,246 @@
+import { PaginationState } from '@ovhcloud/manager-components';
+import {
+  ODS_THEME_COLOR_INTENT,
+  ODS_THEME_TYPOGRAPHY_LEVEL,
+  ODS_THEME_TYPOGRAPHY_SIZE,
+} from '@ovhcloud/ods-common-theming';
+import {
+  ODS_BUTTON_SIZE,
+  ODS_BUTTON_VARIANT,
+  ODS_CHIP_SIZE,
+  ODS_ICON_NAME,
+  ODS_ICON_SIZE,
+  ODS_TEXT_COLOR_HUE,
+} from '@ovhcloud/ods-components';
+import {
+  OsdsButton,
+  OsdsChip,
+  OsdsIcon,
+  OsdsLink,
+  OsdsPagination,
+  OsdsText,
+  OsdsTooltip,
+  OsdsTooltipContent,
+} from '@ovhcloud/ods-components/react';
+import { useTranslation } from 'react-i18next';
+import { TSubnet } from '@/api/data/subnets';
+import { TAggregatedNetwork } from '@/api/data/network';
+
+export type TGlobalRegionsDatagrid = {
+  items: TAggregatedNetwork[];
+  projectUrl: string;
+  totalItems: number;
+  pageCount: number;
+  pagination: PaginationState;
+  onPaginationChange?: (pagination: PaginationState) => void;
+};
+
+export default function GlobalRegionsDatagrid({
+  items,
+  projectUrl,
+  totalItems,
+  pagination,
+  pageCount,
+  onPaginationChange,
+}: TGlobalRegionsDatagrid): JSX.Element {
+  const { t } = useTranslation(['common', 'pagination']);
+
+  const renderBodyRow = (
+    network: TAggregatedNetwork,
+  ): JSX.Element | JSX.Element[] => {
+    const renderText = (
+      text: string | number | JSX.Element,
+      className?: string,
+    ) => (
+      <OsdsText
+        className={`text-center ${className || ''}`}
+        level={ODS_THEME_TYPOGRAPHY_LEVEL.body}
+        color={ODS_THEME_COLOR_INTENT.text}
+        size={ODS_THEME_TYPOGRAPHY_SIZE._400}
+      >
+        {text}
+      </OsdsText>
+    );
+
+    const renderChip = (text: string, color: ODS_THEME_COLOR_INTENT) => (
+      <OsdsChip
+        className="inline-flex m-3"
+        color={color}
+        size={ODS_CHIP_SIZE.sm}
+      >
+        {text}
+      </OsdsChip>
+    );
+
+    const renderTooltipButton = (
+      icon: ODS_ICON_NAME,
+      tooltipContent: string,
+    ) => (
+      <OsdsTooltip>
+        <OsdsButton
+          size={ODS_BUTTON_SIZE.sm}
+          variant={ODS_BUTTON_VARIANT.ghost}
+          color={ODS_THEME_COLOR_INTENT.primary}
+        >
+          <OsdsIcon
+            name={icon}
+            size={ODS_ICON_SIZE.xs}
+            color={ODS_THEME_COLOR_INTENT.primary}
+          />
+        </OsdsButton>
+        <OsdsTooltipContent slot="tooltip-content">
+          {tooltipContent}
+        </OsdsTooltipContent>
+      </OsdsTooltip>
+    );
```

Comment:
what prevents you from creating components instead of functions here ?
They are not context specific so you can just create components and put them outside the component `GlobalRegionDatagrid` for better readability